### PR TITLE
Add a YAML getting started to deploy splunk-otel-collector

### PR DESCRIPTION
**Pull request by flands**
Example is only for gateway mode.

Also update README.

### CODE REVIEW

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        grpc:
+        thrift_binary:
+        thrift_compact:
+        thrift_http:
+    opencensus:
+    otlp:
+      protocols:
+        grpc:
+        http:
+      sapm:
```
Comment (rmfitzpatrick) [inline examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:41]: indentation issue for next 4 receivers

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    port: 13133
+  - name: sapm # Default endpoint for Jaeger gRPC receiver.
+    port: 14250
+  - name: sapm # Default endpoint for Jaeger Thrift receiver.
+    port: 14268
+  - name: sapm # Default endpoint for OpenCensus receiver.
+    port: 55678
+  - name: sapm # Default endpoint for OpenTelemetry receiver.
+    port: 55680
+  - name: sapm # Default endpoint for HTTP Forwarder extension.
```
Comment (rmfitzpatrick) [inline examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:144]: should these all be named sapm?

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  - name: health # Default Health Check port.
+    port: 13133
+  - name: sapm # Default endpoint for Jaeger gRPC receiver.
+    port: 14250
+  - name: sapm # Default endpoint for Jaeger Thrift receiver.
+    port: 14268
+  - name: sapm # Default endpoint for OpenCensus receiver.
+    port: 55678
+  - name: sapm # Default endpoint for OpenTelemetry receiver.
+    port: 55680
```
Comment (rmfitzpatrick) [inline examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:143]: from otlp receiver docs: ```default = 0.0.0.0:4317 for grpc protocol, 0.0.0.0:55681 http protocol```

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml (hunk 2 above)
Comment (flands) [reply examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:144]: nope :)

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml (hunk 3 above)
Comment (flands) [reply examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:143]: yeah, this needs to be updated everywhere...

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml (hunk 1 above)
Comment (flands) [reply examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:41]: nice catch!

File: examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+      jaeger:
+        protocols:
+          grpc:
+          thrift_binary:
+          thrift_compact:
+          thrift_http:
+      opencensus:
+      otlp:
+        protocols:
+          grpc:
```
Comment (rmfitzpatrick) [inline examples/kubernetes-yaml/splunk-otel-collector-gateway.yaml:39]: nit, no associated service/container port entry

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**